src/js/server/lib/hasMailerHandler.js

```diff
@@ -0,0 +1,97 @@
+/*
+
+ `gpii.express.user.api.hasMailHandler` is an instance of `gpii.express.handler` that has an additional invoker to
+ send outgoing mail.  The underlying mailer uses `gpii-handlebars` to render the content, and expects to be passed
+ raw configuration data, which we generate using our data and rules, as follows:
+
+     1. `mailOptions` is a map of valid configuration options for `nodemailer-smtp-transport`.  These options are
+        generated using `options.rules.mailOptions`.
+
+     2. `templateContext` represents data to expose to handlebars during template rendering.  This is generated using
+        `options.rules.mailTemplateContext`.
+
+ The transformation of both sets of rules uses the component itself as its source data, so you have access to any
+ `option` data, all `model` data, and to the implicit `request` object.
+
+To use this component, add it to your list of `gradeNames` and then call the `sendMessage` invoker either from code
+or an invoker definition.
+
+The user won't receive a response until the message has been sent.  If sending occurs without error, the contents of
+`options.messages.success` will be sent to the end user.  If an error occurs, the contents of `options.messages.error`
+will be sent instead.
+
+Note that the mail output is sent using the mailer, and that you must have a `templateDir` and template keys set as
+per that documentation.
+
+ */
+"use strict";
+var fluid  = fluid || require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+
+require("gpii-express");
+
+fluid.registerNamespace("gpii.express.user.api.hasMailHandler");
```

Comment:
I think the modern tendency is to move towards the use of nouns rather than verbs in entity names - how about something like "withMailHandler" or "mailHandlerMixin"?